tmt/steps/report/junit.py

```
@@ -58,40 +42,170 @@ def duration_to_seconds(duration: Optional[str]) -> Optional[int]:
+    :param report: instance of a ReportPlugin.
+    :param flavor: name of a JUnit flavor to generate.
+    :param template_path: if set, the provided template will be used instead of
+        a pre-defined flavor template. In this case, the ``flavor`` must be set
+        to ``custom`` value.
+    :pawam extra_variables: if set, these variables get propagated into the
+        Jinja template.
+    """
+
+    # Get the template context for TMT results
+    results_context = results_context or ResultsContext(report.step.plan.execute.results())
 
-    for result in report.step.plan.execute.results():
+    # Prepare the template environment
+    def _read_log(log: Path) -> str:
+        """ Read the contents of a given result log """
         try:
-            main_log = report.step.plan.execute.read(result.log[0])
-        except (IndexError, AttributeError):
-            main_log = None
-        case = junit_xml.TestCase(
-            result.name,
-            classname=None,
-            elapsed_sec=duration_to_seconds(result.duration))
-
-        if report.data.include_output_log:
-            case.stdout = main_log
-
-        # Map tmt OUTCOME to JUnit states
-        if result.result == tmt.result.ResultOutcome.ERROR:
-            case.add_error_info(result.result.value, output=result.failures(main_log))
-        elif result.result == tmt.result.ResultOutcome.FAIL:
-            case.add_failure_info(result.result.value, output=result.failures(main_log))
-        elif result.result == tmt.result.ResultOutcome.INFO:
-            case.add_skipped_info(result.result.value, output=result.failures(main_log))
-        elif result.result == tmt.result.ResultOutcome.WARN:
-            case.add_error_info(result.result.value, output=result.failures(main_log))
-        # Passed state is the default
-        suite.test_cases.append(case)
-
-    return suite
+            return str(report.step.plan.execute.read(log))
+        except AttributeError:
+            return ''
```

Is it safe to let this slip silently? When can the `read()` end up with an exception?

---

The old implementation was catching `(IndexError, AttributeError)` and it was failing silently setting the `main_log = None`, which is why I was handling it in a similar way:
- https://github.com/teemtee/tmt/blob/c5c214a88af4fb9bf03940386e6df90b0724cf40/tmt/steps/report/junit.py#L71

I'm not entirely sure how to address this. I think I'm fine with handling it more transparently by simply using `return str(phase.step.plan.execute.read(log))` and not catching any exceptions.

---

I decided to just catch the `tmt.utils.FileError` here. Hopefully, this should be enough. Do you agree? Or should I for some reason catch also the `AttributeError`?

---

Ok, let's see whether we can improve it someday.